openedx/core/djangoapps/courseware_api/urls.py

```diff
@@ -8,11 +8,13 @@
 
 from openedx.core.djangoapps.courseware_api import views
 
+SEQUENCE_KEY_OR_HASH_PATTERN = r'(?P<usage_key_or_hash>.+)'
+
 urlpatterns = [
     url(fr'^course/{settings.COURSE_KEY_PATTERN}',
         views.CoursewareInformation.as_view(),
         name="courseware-api"),
-    url(fr'^sequence/{settings.USAGE_KEY_PATTERN}',
+    url(fr'^sequence/{SEQUENCE_KEY_OR_HASH_PATTERN}',
```

Comment:
The view code currently does parsing to determine which this actually is, but can you do it at this layer instead? So keep the existing `settings.USAGE_KEY_PATTERN`, and add a new `url()` function that points to the same view, but with the pattern for the base64 encoded version (which you know is going to be some range of chars over exactly some length), and appropriate kwargs to do the differentiation?

Reply:
Great idea, will do.